tests/infrastructure/golden_images/update_boot_source/test_boot_sources_vm.py

```diff
@@ -0,0 +1,65 @@
+import logging
+import re
+
+import pytest
+
+from utilities.constants import OS_FLAVOR_RHEL
+from utilities.infra import validate_os_info_vmi_vs_linux_os
+from utilities.storage import data_volume_template_with_source_ref_dict
+from utilities.virt import VirtualMachineForTests, running_vm
+
+LOGGER = logging.getLogger(__name__)
+
+
+def assert_os_mismatch_vm(vm, boot_source_data, fedora_version):
```

Comment:
Please add typing for all utilities. Utilities should be in utils.py, not in test file or conftest file.